### ISSUE TITLE
[FEATURE] Add new best practice rule for `<details>` and `<summary>`

### ISSUE THREAD

**Issue by aaronadamsCA** [yeonjuan/html-eslint#518]
**Describe the bug**

Interactive elements should be allowed inside `<details>`.

I realize this is weird, given the HTML spec says `<details>` is interactive. But apparently it's only `<summary>` that's meant to be:

> https://github.com/whatwg/html/issues/11342
>
> `<details>` is not interactive. The `<summary>` header is interactive, and thus [#2272](https://github.com/whatwg/html/issues/2272), but details itself is not.

We use links inside disclosures too often to `eslint-ignore` all of them, but this rule is too valuable to turn off completely, so we've reduced it to warn for now.

**To Reproduce**

I copied this example directly from the HTML spec ([source](https://html.spec.whatwg.org/multipage/interactive-elements.html#the-details-element)):

```html
<details>
 <summary><label for=fn>Name & Extension:</label></summary>
 <p><input type=text id=fn name=fn value="Pillar Magazine.pdf">
 <p><label><input type=checkbox name=ext checked> Hide extension</label>
</details>
```

**Expected behavior**

I think this might be the simplest solution:

1. Update this rule to replace `details` with `summary`.

   In the above example, this would still error on `<label>` inside `<summary>`. Chrome 140 recently added a warning for interactive elements inside `<summary>` (https://issues.chromium.org/issues/427172874), so I think that error still makes sense as the standard continues to evolve.

2. Add new best practice rule `require-details-summary` to guard against this:

   ```html
   <button>
     <details>Button</details>
   </button>
   ```

   If I'm reading the spec correctly, the content of `<details>` should always begin with `<summary>`. So if one rule requires `<summary>` inside `<details>`, and another rule bans `<summary>` inside `<button>`, then between the two rules you should be covered.

**Comment by yeonjuan** [comment]
Hi @aaronadamsCA Thank you for your report. Let me fix it.

**Comment by yeonjuan** [comment]
The bug has been fixed and released. I will keep this issue open as there is still a proposal for implementing a new rule. 👍
